scene/animation/animation_node_extension.h

```diff
@@ -0,0 +1,55 @@
+/**************************************************************************/
+/*  animation_node_extension.h                                            */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef ANIMATION_NODE_EXTENSION_H
+#define ANIMATION_NODE_EXTENSION_H
+
+#include "scene/animation/animation_tree.h"
+
+class AnimationNodeExtension : public AnimationRootNode {
+	GDCLASS(AnimationNodeExtension, AnimationRootNode);
```

Comment:
Related to the comment above, AnimationNodeExtention should replace AnimationNode, not AnimationRootNode.

The AnimationRootNode does not need an input port and can be placed in a StateMachine. If not, it is because that would imply that an input port is required.

I guess you want it to be the AnimationRootNode for placing it in the BlendSpace or StateMachine purposes, but if you replace it and replace the AnimationNode with the AnimationNodeExtention, the AnimationRootNode will extend it, thus solving this classification problem.

Reply:
Btw, was this comment overlooked?

Reply:
Yes I think if it is an Extension, we will need two Extensions, AnimationRootNodeExtension and AnimationNodeSyncExtension. If it is not an Extension (by replacement), then only the AnimationNode needs to be replaced. cc @GuilhermeGSousa

Reply:
https://github.com/godotengine/godot/pull/100416